build/document-extractor.js

```diff
@@ -251,7 +266,16 @@ function _addSingleSectionBCD($) {
     }
   }
 
-  const dataQuery = $.find("div.bc-data").attr("id");
+  let dataQuery = null;
+  let specialSectionType = null;
+  if ($.find("div.bc-data").length) {
```

Comment:
Given that the code becomes lengthier and lengthier, perhaps worth to have an early check and then branch out into different function calls?

Personally, I'd consider this function to have too many variables, which would lead me to think, that it tries to do too many things => harder to test and maintain.